### ISSUE TITLE
Where can we download the compiled COBie jar files that can be placed in the plugin directory for BIMServer 1.3.0

### ISSUE THREAD

**Issue by webdada** [opensourceBIM/COBie-plugins#4]
Thanks.

Regards,

Sam Wu
swu@samssoft.com

**Comment by OhmSweetOhm** [comment]
I need to push an update to the code base and rebuild the binaries.  I'll holler back when the update is complete.

**Comment by webdada** [comment]
Thanks - when I build the code in Eclipse, it is missing a directory called
EMF, and also the build.xml etc in the directory, there are quite a few
other missing pieces.

So that is why I asked for the cobie jar file, to be placed in the
BIMServer plugins directory.

Regards,

Sam
Director, SAMSSOFT LLC
Reston, VA

On Fri, Aug 22, 2014 at 9:44 AM, OhmSweetOhm notifications@github.com
wrote:

> I need to push an update to the code base and rebuild the binaries. I'll
> holler back when the update is complete.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/opensourceBIM/COBie-plugins/issues/4#issuecomment-53061672
> .

## 

Sam
dada.sam@gmail.com

**Comment by OhmSweetOhm** [comment]
that EMF dependency is deprecated

I'm updating all of this and will push soon.

**Comment by webdada** [comment]
Thank you so much OhmSweetOhm ! Looking forward to work with you on this plugin and other BIMServer.org related projects. 

Sam

**Comment by OhmSweetOhm** [comment]
The plugin jar is now available along with the updated codebase.

The COBieDeserializer seems to be broken though.  Something's up with the way BIMServer is handling the geometry (or non geometry) entities.  Perhaps it's not liking the Coordinate sheet deserializer - which probably should be deprecated anyway.

**Comment by webdada** [comment]
Thank you OhmSweetOhm. 
My team member Bill is working with me on the BIMServer integration. 
Now we would like to use COBie plugin to export a COBie file from IFC file. Can you please share some code examples of how to export COBie file?

For example: the following code would return Null.
SSerializerPluginConfiguration serializer =   client.getBimsie1ServiceInterface().getSerializerByContentType("application/COBIE");

Regards,

Sam 
SAMSSOFT LLC

---

Bill

2:00 PM (2 hours ago)

to me
While using BimSie1 API to download cobie from a IFC in bimServer, I use the code to retrieve serilizer first. However the content type "application/COBIE" is not working. Why I use "application/COBIE"? because "application/ifc" ,"application/json" ,"application/ifcxml"  were working for IFC, JSON, IFCXML download well, so I assume "application/COBIE" is for cobie download. I really could not find this assumption is correct.

There is no much example on how to download Cobie using BimSie1. So I was trying various strings such as "application/COBie","application/cobie","application/COBIE", but they were not working.

SSerializerPluginConfiguration serializer = client.getBimsie1ServiceInterface().getSerializerByContentType("application/COBIE");

With content Type = application/COBIE, serializer returns Null.

So the question here:

 Is "application/COBIE" incorrect content type?

https://opensourcebim.zendesk.com/hc/communities/public/questions/200799105-Unable-to-download-ifc-as-cobie-in-bimserver1-3-0-version

**Comment by OhmSweetOhm** [comment]
I never use the getSerializerByContentType method.  I prefer fetching the serializers by name - though the names can be re-configured in the server settings.  Several serializers may have the same content type (e.g. xml), so this seems like an unreliable way to fetch the serializers.  That's just my opinion though.

The cobie spreadsheet file in the cobie serializer is an xml file that can be opened in Excel.  If  I'm not mistaken, it is configured as application/.xml.

Sorry I can't be of much help here.

**Comment by webdada** [comment]
Thank you so much !

I am copying Bill Yuan in this email, so if he has further questions we
will follow up.

Regards,

Sam Wu
SAMSSOFT LLC

On Wed, Sep 3, 2014 at 4:12 PM, OhmSweetOhm notifications@github.com
wrote:

> I never use the getSerializerByContentType method. I prefer fetching the
> serializers by name - though the names can be re-configured in the server
> settings. Several serializers may have the same content type (e.g. xml), so
> this seems like an unreliable way to fetch the serializers. That's just my
> opinion though.
> 
> The cobie spreadsheet file in the cobie serializer is an xml file that can
> be opened in Excel. If I'm not mistaken, it is configured as
> application/.xml.
> 
> Sorry I can't be of much help here.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/opensourceBIM/COBie-plugins/issues/4#issuecomment-54358241
> .

## 

Sam
dada.sam@gmail.com

**Comment by webdada** [comment]
Thanks very much for the help! I agreed it is not reliable way to use
"application/COBIE". At one point, "application/COBIE" was working but for
some reason I could not repeat it.  I also tried to use
getSerializerByClassName
with a parameter of plugin implementation class name. But
"application/.xml" definitely is worth trying.

Bill

On Wed, Sep 3, 2014 at 4:53 PM, Sam Wu dada.sam@gmail.com wrote:

> Thank you so much !
> 
> I am copying Bill Yuan in this email, so if he has further questions we
> will follow up.
> 
> Regards,
> 
> Sam Wu
> SAMSSOFT LLC
> 
> On Wed, Sep 3, 2014 at 4:12 PM, OhmSweetOhm notifications@github.com
> wrote:
> 
> > I never use the getSerializerByContentType method. I prefer fetching the
> > serializers by name - though the names can be re-configured in the server
> > settings. Several serializers may have the same content type (e.g. xml), so
> > this seems like an unreliable way to fetch the serializers. That's just my
> > opinion though.
> > 
> > The cobie spreadsheet file in the cobie serializer is an xml file that
> > can be opened in Excel. If I'm not mistaken, it is configured as
> > application/.xml.
> > 
> > Sorry I can't be of much help here.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/opensourceBIM/COBie-plugins/issues/4#issuecomment-54358241
> > .
> 
> ## 
> 
> Sam
> dada.sam@gmail.com

## 

Bill Yuan

**Comment by webdada** [comment]
Hello Leo/Ruben,
The download of Cobie in bimserver 1.3 is still not working as the output
of serializer is null for various types:
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/COBie"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/COBIE"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/cobie"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("APPLICATION/COBIE"));

out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("COBIE"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/COBIE__NEW"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/.xml"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/xml"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/"));.

 I also tried
client.getBimsie1ServiceInterface().getSerializerByName("org.bimserver.scenejs.SceneJSSerializerPlugin
is null.

I put the cobie plugin.jar into the bimserver web_info, plugin folder.

I searched the example on github and wiki, could not find a code on how to
download as cobie.

A further help is needed, I will try other options.
thanks,
Bill

null null null null null null null null
 JsonBimServerClientFactory factory = new JsonBimServerClientFactory(
"http://localhost:80/bimserver13");
try {
BimServerClient client = factory
.create(new UsernamePasswordAuthenticationInfo(
userName, passWord));

Bimsie1ServiceInterface serviceInterface = client
.getBimsie1ServiceInterface();
Bimsie1LowLevelInterface lowLeveInterface = client
.getBimsie1LowLevelInterface();

out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/COBie"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/COBIE"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/cobie"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("APPLICATION/COBIE"));

out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("COBIE"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/COBIE__NEW"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/.xml"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/xml"));
out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/"));
//out.println(new java.util.Date()+ "---serializer=" + serializer);
//SSerializerPluginConfiguration serializer =
client.getBimsie1ServiceInterface().getSerializerByName("org.bimserver.scenejs.SceneJSSerializerPlugin");

//client.download(Long.parseLong("524291"), serializer.getOid(),new
File("C:\Dev\GeneratedSoapClient\\bimserver\output."+
oid+"."+downloadType));
//out.println("downloadType is download for:" + downloadType);
} catch (Exception e) {
out.println(" exception!" + e);
e.printStackTrace();
}
%>

On Wed, Sep 3, 2014 at 4:53 PM, Sam Wu dada.sam@gmail.com wrote:

> Thank you so much !
> 
> I am copying Bill Yuan in this email, so if he has further questions we
> will follow up.
> 
> Regards,
> 
> Sam Wu
> SAMSSOFT LLC
> 
> On Wed, Sep 3, 2014 at 4:12 PM, OhmSweetOhm notifications@github.com
> wrote:
> 
> > I never use the getSerializerByContentType method. I prefer fetching the
> > serializers by name - though the names can be re-configured in the server
> > settings. Several serializers may have the same content type (e.g. xml), so
> > this seems like an unreliable way to fetch the serializers. That's just my
> > opinion though.
> > 
> > The cobie spreadsheet file in the cobie serializer is an xml file that
> > can be opened in Excel. If I'm not mistaken, it is configured as
> > application/.xml.
> > 
> > Sorry I can't be of much help here.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/opensourceBIM/COBie-plugins/issues/4#issuecomment-54358241
> > .
> 
> ## 
> 
> Sam
> dada.sam@gmail.com

## 

Bill Yuan

**Comment by billwyuan** [comment]
OhmSweetOhm, 

I have tested the following methods with content type after putting the Cobie1.3.jar into plugin folder in bimserver, but got null in all outputs. Is there anything I need to configure on BIMServer side? Both Sam and myself have been struggling this for a few weeks. Your assistance is greatly appreciated.
        out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/COBie"));
        out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/COBIE"));
        out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/cobie"));
        out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("APPLICATION/COBIE"));

```
    out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("COBIE"));
    out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/COBIE__NEW"));
    out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/.xml"));
    out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/xml"));
    out.println(client.getBimsie1ServiceInterface().getSerializerByContentType("application/"));
```

**Comment by OhmSweetOhm** [comment]
Have you tried pluginInterface.getSerializerByPluginClassName?

You would have to give it the fully qualified class name of the serializerplugin

e.g. pluginInterface.getSerializerByPluginClassName("org.erdc.cobie.plugins.serializers.COBieSerializerPlugin");

**Comment by webdada** [comment]
Sent from my iPhone

On Sep 9, 2014, at 4:33 PM, OhmSweetOhm notifications@github.com wrote:

I never tested the getSerializerByContentType method

I find them by DefaultName...for example...
private Deserializer initializeDeserializerByName(String name) throws
PluginException
{
Deserializer deserializer = null;
for (DeserializerPlugin deserializerPlugin :
pluginManager.getAllDeserializerPlugins(true))
{
if (deserializerPlugin.getDefaultName().equals(name))
{
DeserializerPlugin plugin = deserializerPlugin;
if (!plugin.isInitialized())
{
plugin.init(pluginManager);
}
deserializer = plugin.createDeserializer(null);
}
}
return deserializer;
}

**Comment by billwyuan** [comment]
Great, I got the serializer id. it is worth trying. I will make a download using it.
Thanks!

Best regards,
Bill Yuan

**Comment by TCRAKN** [comment]
Hello all,

Is there a compiled version of COBie.jar available for BIMserver 1.4?

I'm not sure where to start with Eclipse and the COBie Plugin source files, to produce my own compiled file.

TIA
Abdul-Khaliq

**Comment by OhmSweetOhm** [comment]
TCRAKN, yes there is a 1.4 version that was prepared at the end of July 29.  Download it in the releases section.
